src/components/Basic/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,86 @@
+import { createSlot, createSlots } from "@components/BaseComponent/Slots";
+import { ComponentProps } from "@components/types/ComponentProps";
+import { Accessor, createSignal, onMount, ParentComponent, ParentProps, Show } from "solid-js";
+import styles from './Checkbox.module.css';
+import { BaseComponent } from "@components/BaseComponent/BaseComponent";
+import CheckboxControl, { CheckboxSlotProps } from "./CheckboxControl";
+
+export interface CheckboxRef {
+    checked: Accessor<boolean>,
+    value: any,
+    toggle: (e?: MouseEvent) => void,
+    element: HTMLDivElement,
+}
+
+interface CheckBoxProps extends ComponentProps {
+    value?: any
+    checked?: boolean
+    disabled?: boolean
+    onChange?: (checked: boolean) => void;
+}
+
+interface LabelProps extends ParentProps  {
+    before?: boolean
+}
+
+const { useSlots, withSlots } = createSlots({
+    Label: createSlot<LabelProps>(),
+    Control: createSlot<CheckboxSlotProps>(),
+    Indicator: createSlot<CheckboxSlotProps>(),
+});
+
+const Checkbox: ParentComponent<CheckBoxProps> = (props) => {
+    const { slots } = useSlots();
+    const [checked, setChecked] = createSignal(props.checked ?? false);
+    const isBefore = slots.Label?.before;
+    let element!: HTMLDivElement;
+
+    props.componentClasses = `${styles.Checkbox} ${props.disabled ? styles.Disabled : ''}`;
+ 
+    const toggle = (e?: MouseEvent) => {
+        if (props.disabled) return;
+
+        setChecked(prev => !prev);
+        props.click?.(e as MouseEvent);
+        props.onChange?.(checked())
+    }
+
+    onMount(() => {
+        if (!props.ref || !element) return;
+
+        (props.ref as unknown as (ref: any) => void)({
+            checked,
+            value: props.value,
+            toggle,
```

Comment:
Why do we need the toggle method in the ref? Maybe expose `setChecked` instead so the developer can set `true` or `false` and check/uncheck the checkbox instead of toggling it.